src/templates/ecosystem-page/ProjectsAndTools.tsx

```diff
@@ -0,0 +1,133 @@
+import { FC, useEffect, useState } from "react"
+import { SimpleGrid, Stack, Flex } from "@chakra-ui/react"
+import { ImageProps } from "../../components"
+import SortDropdown, { SortOption } from "../../components/SortDropdown"
+import FilterMenu from "../../components/FilterMenu"
+import { ProjectsAndToolsCard } from "./ProjectsAndToolsCard"
+import Pagination from "../../components/Pagination"
+import { convertDateToTimestamp } from "../../utils/date"
+
+const ITEMS_PER_PAGE = 6
+
+export enum Category {
+  ALL = "all",
+  PROJECT = "project",
+  TOOL = "tool",
+  INFRASTRUCTURE = "infrastructure",
+  INTEGRATION = "integration",
+}
+
+export interface CardButton {
+  label: string
+  url: string
+  variant: string
+}
+
+export interface CardCategory {
+  label: string
+}
+
+export interface ProjectsAndToolsCardProps {
+  image: ImageProps
+  title: string
+  description: string
+  categories: CardCategory[]
+  date: string
+  buttons: CardButton[]
+}
+
+const ProjectsAndTools: FC<{ cards: ProjectsAndToolsCardProps[] }> = ({
+  cards,
+}) => {
+  const [totalNumberOfCards, setTotalNumberOfCards] = useState<number>(0)
+  const [paginatedCards, setPaginatedCards] = useState<
+    ProjectsAndToolsCardProps[]
+  >([])
+  const [activeCategory, setActiveCategory] = useState<string | null>(
+    Category.ALL
+  )
+  const [sortOption, setSortOption] = useState<SortOption>()
+
+  const [currentPage, setCurrentPage] = useState(1)
+  const totalPages = Math.ceil(totalNumberOfCards / ITEMS_PER_PAGE)
+
+  const filterAndSortCards = () => {
+    const tempCards =
+      activeCategory === Category.ALL
+        ? cards
+        : cards.filter((card) =>
+            card.categories.some(
+              (category) => category.label === activeCategory
+            )
+          )
+
+    // Sorting logic
+    return tempCards.sort((a, b) => {
+      const aDate = convertDateToTimestamp(a.date)
+      const bDate = convertDateToTimestamp(b.date)
+
+      switch (sortOption) {
+        case SortOption.TITLE_ASC:
+          return a.title.localeCompare(b.title)
+        case SortOption.TITLE_DESC:
+          return b.title.localeCompare(a.title)
+        case SortOption.TIMESTAMP_ASC:
+          return aDate - bDate
+        case SortOption.TIMESTAMP_DESC:
+          return bDate - aDate
+        default:
+          return 0
+      }
+    })
+  }
+
+  useEffect(() => {
+    const sortedAndFilteredCards = filterAndSortCards()
+
+    // Set total number of cards instead of the entire array
```

Comment:
Non-blocking:

I think we can remove this comment. I don't think it's necessary to point out that we set the total number of cards instead of the entire array here. I believe comments should refer to the current state of the code and the code itself (at it's current state) doesn't indicate that we were setting the entire array in the past. So the comment might be confusing as it refers to the past commits that are no longer actual. We can put such information in the commit message if needed.

Reply:
Addressed in https://github.com/threshold-network/website/pull/94/commits/3e566e97f8b9c112b35d141c2b494100d9ee706c